### PR TITLE
[Core] Remove multiple core workers in one process 2/n

### PR DESCRIPTION
**Pull request by jjyao**
## Why are these changes needed?
Remove some legacy code related to multiple core workers in one process since we now no longer support it.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/raylet/worker_pool.cc [hunk 1]
```diff
@@ -1063,126 +1061,73 @@ void WorkerPool::TryKillingIdleWorkers() {
       RAY_LOG(DEBUG) << "idle worker is already dead. Not going to kill worker "
                      << idle_worker->WorkerId();
       // This worker has already been killed.
-      // This is possible because a Java worker process may hold multiple workers.
```
Comment (cadedaniel) [inline src/ray/raylet/worker_pool.cc:1066]: nit: improve comment to say when this is possible? suggestion below which may not be correct
> This is possible if the idle_worker has been killed but hasn't been removed from the idle list yet.

**Review by cadedaniel — APPROVED** [review]
this is awesome, makes my life easier :)

**Review by cadedaniel — CHANGES_REQUESTED** [review]
ah meant to only comment. tests still running

**Comment by jjyao** [comment]
Failed tests are unrelated.

**Review by cadedaniel — APPROVED** [review]

**Comment by jovany-wang** [comment]
Nice PR!

**Pull request merged**